code/modules/mob/living/simple_animal/hostile/necro.dm

```diff
@@ -96,11 +130,313 @@
 		ticker.mode.update_necro_icons_added(Controller)
 		ticker.mode.update_all_necro_icons()
 		ticker.mode.risen.Add(Controller)
-	name += " ([rand(1,1000)])"
 
-/mob/living/simple_animal/hostile/necro/copy/ListTargets()
-	. = ..()
-	return . - creator
+	if(name == initial(name) && !unique_name)
+		name += " ([rand(1,1000)])"
+
+/mob/living/simple_animal/hostile/necro/zombie/Life()
+	..()
+	/*TODO
+	First, check if the zombie can potentially evolve
+	Have the zombie move to a corpse and start chewing at it
+	If neither of these things are applicable, break some lights to set the mood
+	Otherwise, start wandering and bust down some doors to find more food
+	*/
+	if(!isUnconscious())
+		if(stance == HOSTILE_STANCE_IDLE) //Not doing anything at the time
+			var/list/can_see = view(src, vision_range)
+			if(!busy)
+				if(can_evolve)//Can we evolve, and have we fed
+					busy = EVOLVING
+					check_evolve()
+					busy = 0
+				if((health < maxHealth) || (maxHealth < health_cap)) //Is there something to eat in range?
+					for(var/mob/living/carbon/human/C in can_see) //If so, chow down
+						if(C.isDead() && !busy && check_edibility(C))
```

Comment:
It's plotting a new path to every human it can see in this loop because the loop doesn't break. You may want to have the loop find the closest human and then plot a course to him, or something else.